Library/Homebrew/cmd/--env.rb

```diff
@@ -3,19 +3,37 @@
 
 require "extend/ENV"
 require "build_environment"
+require "utils/shell"
 
 module Homebrew
+  def explicit_shell
+    shell = ARGV.value("shell")
+    return :noshell if shell == "none"
```

Comment:
With no `--shell`, I was going to keep the behavior the same as it is now. If stdout is a tty, `brew --env` outputs in the `key: value` format.

current behavior:

stdout is tty

    $ brew --env
    CMAKE_PREFIX_PATH: /usr/local
    ...
    PATH: /usr/local/Library/ENV/4.3:/usr/bin:/bin:/usr/sbin:/sbin

stdout is not tty

    $ brew --env | cat
    export CC="clang"
    ...
    export ACLOCAL_PATH="/usr/local/share/aclocal"

There is currently no way to force `brew --env` to print the environment variables and their values in the `key: value` format, stdout has to really be a tty. That's why I was thinking of adding something like `--shell=none`.